contracts/interchain-token-service/src/contract.rs

```diff
@@ -378,17 +387,7 @@ impl InterchainTokenService {
     ) -> Result<Bytes, ContractError> {
         // Note: ITS Hub chain as the actual destination chain for the messsage isn't supported
         ensure!(
-            destination_chain != Self::its_hub_chain_name(env),
-            ContractError::UntrustedChain
-        );
-
-        let Some(destination_address) = Self::trusted_address(env, destination_chain.clone())
-        else {
-            return Err(ContractError::UntrustedChain);
-        };
-
-        ensure!(
-            destination_address == Self::its_hub_routing_identifier(env),
+            Self::is_trusted_chain(env, destination_chain.clone()),
```

Comment:
get_call_params doesn't seem very useful anymore. Move its code to the caller instead?